confmap/provider/secretsmanagerprovider/provider.go

```diff
@@ -40,29 +47,56 @@ func newWithSettings(_ confmap.ProviderSettings) confmap.Provider {
 //
 // Deprecated: [v0.100.0] Use NewFactory() instead.
 func New() confmap.Provider {
-	return &provider{}
+	return &provider{client: nil}
 }
 
 func (provider *provider) Retrieve(ctx context.Context, uri string, _ confmap.WatcherFunc) (*confmap.Retrieved, error) {
 	if !strings.HasPrefix(uri, schemeName+":") {
 		return nil, fmt.Errorf("%q uri is not supported by %q provider", uri, schemeName)
 	}
 
-	secretArn := strings.Replace(uri, schemeName+":", "", 1)
+	// initialize the secrets manager client in the first call of Retrieve
+	if provider.client == nil {
+		cfg, err := config.LoadDefaultConfig(context.Background())
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to load configurations to initialize an AWS SDK client, error: %w", err)
+		}
+
+		provider.client = secretsmanager.NewFromConfig(cfg)
```

Comment:
Is this client stateful, does it need to be closed later?

Reply:
Client in AWS SDK v2 don't need to be closed.